pydra/engine/core.py

```diff
@@ -522,6 +543,7 @@ def add(self, task):
         """adding a task to the workflow"""
         if not is_task(task):
             raise ValueError("Unknown workflow element: {!r}".format(task))
+        task._reset()
```

Comment:
does this need to happen at every `add` call? or could this be done just prior to `run` or perhaps more preferably at the end of a `run`?